### PR TITLE
Use CliWrap instead of System.Diagnostics

### PR DESCRIPTION
**Pull request by CodeMouse92**
Employs CliWrap, as per comment in #51.

### CODE REVIEW

File: Corgibytes.Freshli.Cli/Functionality/Git.cs
Comment (mscottford) [inline Corgibytes.Freshli.Cli/Functionality/Git.cs:115]: Calling `Result` on the next line implies `Wait()`, but I see little harm in being explicit here.

**Review by mscottford — APPROVED** [review]
This looks good!

File: Corgibytes.Freshli.Cli/Functionality/Git.cs
Comment (mscottford) [inline Corgibytes.Freshli.Cli/Functionality/Git.cs:112]: We need to add `using` here to make sure that the `CommandTask<T>` instance is properly cleaned up, because it implements `IDisposable`.

```suggestion
        using var command = CliWrap.Cli.Wrap(gitPath).WithArguments(
                args => args
                    .Add("clone")
                    .Add(Url)
                    .Add('.')
            )
            .WithValidation(CommandResultValidation.None)
            .WithWorkingDirectory(Directory.FullName)
            .WithStandardErrorPipe(PipeTarget.ToStringBuilder(stdErrBuffer));
```

File: Corgibytes.Freshli.Cli/Functionality/Git.cs
Comment (mscottford) [inline Corgibytes.Freshli.Cli/Functionality/Git.cs:127]: `using` is also needed here.

File: Corgibytes.Freshli.Cli/Functionality/Git.cs
Comment (mscottford) [inline Corgibytes.Freshli.Cli/Functionality/Git.cs:148]: `using` is also needed here.

**Review by mscottford — CHANGES_REQUESTED** [review]
Eek! I noticed something after I'd already submitted an "approved" review. Sorry for the confusion.

File: Corgibytes.Freshli.Cli/Functionality/Git.cs
Comment (CodeMouse92) [reply Corgibytes.Freshli.Cli/Functionality/Git.cs:115]: It *wasn't* implying it, which is why I mentioned in Slack about the race condition.

File: Corgibytes.Freshli.Cli/Functionality/Git.cs
Comment (CodeMouse92) [reply Corgibytes.Freshli.Cli/Functionality/Git.cs:112]: Does not work. If you make this change in Jetbrains, it will immediately alert you with the following:

```
Type 'CliWrap.Command' used in a 'using' declaration must be implicitly convertible to 'System.IDisposable'|
```

Comment (mscottford) [reply Corgibytes.Freshli.Cli/Functionality/Git.cs:112]: I liked the solution that you came up with. Thanks!

**Review by mscottford — APPROVED** [review]
Approved pending conflict resolution with the `main` branch.

**Pull request merged**
